AI/VCAI/AIUtility.h

```diff
@@ -166,16 +171,18 @@ bool shouldVisit(HeroPtr h, const CGObjectInstance * obj);
 
 ui64 evaluateDanger(const CGObjectInstance * obj);
 ui64 evaluateDanger(crint3 tile, const CGHeroInstance * visitor);
-bool isSafeToVisit(HeroPtr h, crint3 tile);
 bool isObjectRemovable(const CGObjectInstance * obj); //FIXME FIXME: move logic to object property!
+bool isSafeToVisit(HeroPtr h, uint64_t dangerStrength);
+bool isSafeToVisit(HeroPtr h, crint3 tile);

```

Comment:
think it would be better if it will be HeroPtr member method and then marked as const

Reply:
As far as I understood HeroPtr is kind of smart pointer? If we will start adding some methods there we will mix it a bit. Especially when calling CGHeroInstance we will use -> and when calling isSafeToVisit it will be just a dot. It looks weird. It might be a responsibility of some BattleManager or something. Do you want to get rid of AIUtility or do you want to have some AIHeroInstance with additional behavior and functionality?

Reply:
huh, I just don't like these globals there, any way to get rid of them?